### DISCUSSION TITLE
No ability to add custom submenu icon in menubar

### DISCUSSION THREAD

**Discussion by AlexanderLevchenko** [primefaces/community#738] [PrimeNG] [answered]
Hi, colleagues!
Since primeng **v16.2** we faced with a problem:
we have no ability to add custom submenu icon in **menubar**

![image](https://github.com/primefaces/community/assets/10401131/88cb8ee4-67b9-421b-b7ed-f0e6a114b1ac)

This is because you are trying to get **submenuIconTemplate** from **menubar**, but appropriate **viewChild** for **menubar** is **menubarViewChild**
![image](https://github.com/primefaces/community/assets/10401131/e3bc13a5-f63b-42c3-a4d4-06b61176b1f3)

![image](https://github.com/primefaces/community/assets/10401131/c92155c4-89bf-4b3a-9284-81c36bffb5c2)

I believe, **submenuIconTemplate** should be retrieved from **menubarViewChild** instead of **menubar** in the template

This is not reproducible in **v16.0.2**
Could you please help on this?
Thank you in advance!

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks a lot for reporting the issue, this issue is fixed and the fix will be available by upcoming release tomorrow https://github.com/primefaces/primeng/issues/14430

**Reply by AlexanderLevchenko** [reply]
Thank you a lot!
